cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -742,17 +746,17 @@ public void testDelete() throws Exception {
     response = doDelete(getVersionedAPIPath(
       String.format("apps/%s/versions/%s", appId.getApplication(), appId.getVersion()),
       Constants.Gateway.API_VERSION_3_TOKEN, TEST_NAMESPACE2));
-    Assert.assertEquals(404, response.getResponseCode());
+    Assert.assertEquals(500, response.getResponseCode());
```

Comment:
don't modify existing tests for this, it means we're losing test coverage for the old path.

There should be a new test for the disabled behavior.